### PR TITLE
fix: clear invalid `version_log`

### PR DESCRIPTION
**Pull request by KKould**
https://github.com/tonbo-io/tonbo/issues/142#issuecomment-2339819930  @crwen I implemented the content in the comment in this pr

### CODE REVIEW

File: src/version/set.rs [hunk 1]
```diff
@@ -81,16 +81,39 @@ where
...
+        // when there are multiple logs, this means that a downtime occurred during the
+        // `version_log_snap_shot` process, the second newest file has the highest data
+        // integrity, so it is used as the version log, and the older log is deleted first
+        // to avoid midway downtime, which will cause the second newest file to become the
+        // first newest after restart.
+        let mut i = 0;
+        while let Some(result) = log_stream.next().await {
+            let (log, log_id) = result?;
+
+            if i <= 1 {
```
Comment (crwen) [inline src/version/set.rs:102]: `list` sort files in increasing order, so the last one is the newest file?
https://github.com/tonbo-io/tonbo/blob/f4ab2e54af749181a2dc2f104189dd56854b9c5a/src/fs/tokio_fs.rs#L40

Comment (KKould) [reply src/version/set.rs:102]: yep, so I reversed it
https://github.com/tonbo-io/tonbo/blob/f4ab2e54af749181a2dc2f104189dd56854b9c5a/src/version/set.rs#L87

Comment (crwen) [reply src/version/set.rs:102]: Oops, didn't notice that. But it seems that `Ulid::new` will not guarantee monotonic sort order. maybe use `ulid::Generator` instead?
https://docs.rs/ulid/latest/ulid/struct.Ulid.html#method.new

Comment (KKould) [reply src/version/set.rs:102]: You are right, we should be able to open this as a new issue

**Review by crwen — APPROVED** [review]

File: src/version/edit.rs [hunk 2]
```diff
@@ -122,7 +122,7 @@ where
                 let len = u32::decode(reader).await?;
                 VersionEdit::NewLogLength { len }
             }
-            _ => todo!(),
+            i => todo!("{i}"),
```
Comment (ethe) [inline src/version/edit.rs:125]: `unreachable()`?

**Pull request merged**